### PR TITLE
164101934 calenda date pick

### PR DESCRIPTION
**Pull request by jkorpelin**
# Changed
* The logic behind date selections in the calendar component.

### CODE REVIEW

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 1]
```diff
@@ -71,18 +72,22 @@
...
                      inset 0 0 0 2px transparent"})
-     (cond (= (time/format-date-iso-8601 date1) d)
+     (cond (and (= (time/format-date-iso-8601 date1) d) (nil? date2))
+           (style/date1-highlight-style hash-color colors/gray800)
+
+           (= (time/format-date-iso-8601 date1) d)
            (style/date1-highlight-style hash-color)
 
-           (= (time/format-date-iso-8601 date2) d)
+           (= (and (not (nil? date2)) (time/format-date-iso-8601 date2)) d)
```
Comment (mattiuusitalo) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: (some? date2) is same as (not (nil? date2)), is it not?

File: ote/src/cljs/ote/app/controller/transit_visualization.cljs [hunk 2]
```diff
@@ -381,48 +381,49 @@
...
+                    date1 (get-in app [:transit-visualization :compare :date1])
+                    date2 (get-in app [:transit-visualization :compare :date2])
+                    route (get-in app [:transit-visualization :selected-route])
+                    goog-date1 (goog.date.DateTime. date1)
+                    goog-date (goog.date.DateTime. date)
+                    date-after-date1? (t/after? goog-date goog-date1)
+                    earlier-date (if date-after-date1? date1 date)
+                    later-date (if date-after-date1? date date1)]
+                (cond
+                  (or (and date1 date2) (t/equal? goog-date1 goog-date))
```
Comment (mattiuusitalo) [inline ote/src/cljs/ote/app/controller/transit_visualization.cljs]: using a state machine could make this easier to read

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 3]
```diff
@@ -71,18 +72,22 @@
...
            (style/date2-highlight-style hash-color)))))
 
 (defn hover-day [e! date->hash day]
   (e! (tv/->HighlightHash (date->hash (time/format-date day)) day))
   (e! (tv/->DaysToFirstDiff (time/format-date day) date->hash)))
 
-(defn select-day [e! day]
-  (e! (tv/->SelectDateForComparison day)))
+(defn select-day [e! day loading?]
+  (when (not loading?)
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: Nitpicking: `when-not loading?`

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 4]
```diff
@@ -71,18 +72,22 @@
      (when (and prev-week-hash hash (not= hash prev-week-hash) (> day (time/now)))
        {:box-shadow "inset 0 0 0 1px black,
                      inset 0 0 0 2px transparent"})
-     (cond (= (time/format-date-iso-8601 date1) d)
+     (cond (and (= (time/format-date-iso-8601 date1) d) (nil? date2))
+           (style/date1-highlight-style hash-color colors/gray800)
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: Nitpicking: define logical styles `day-neutral`, `day-compare-1st` and `day-compare-2nd`?

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 5]
```diff
@@ -552,36 +557,22 @@
...
-                                                                                                  style/date2-highlight-color-hover)))))}]
-
-       [:h3 "Valittujen päivämäärien väliset muutokset"]
-       [comparison-date-changes compare]]]]))
+                                                           (style/date-highlight-style hash-color
+                                                                                       style/date-highlight-color-hover))}]
+       (when (and (:date1 compare) (:date2 compare))
+         [:div
+          [:h3 "Valittujen päivämäärien väliset muutokset"]
+          [comparison-date-changes compare]])]]]))
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: This is ok, another alternative could be to have `comparison-date-changes`displayed all the time and just clear it's data model to have it displayed empty for a moment, if such UX is desired.
No need to change, as said :)

**Pull request merged**